src/biogeochem/FanMod.F90

```diff
@@ -0,0 +1,1214 @@
+    else
+       fraction_down = 0.0
+       fraction_in = 0.0
+       fraction_runoff = 1.0_r8
+    end if
+    
+  end subroutine partition_to_layer
+
+  subroutine age_pools_soil(ndep, dt, pools, mtan, garbage)
+    !
+    !  Evaluate the N pool "aging" for one time step; moves N mass from the pools
+    !  representing younger to older N patches.
+    !
+    implicit none
+    real(r8), intent(in) :: ndep ! flux of TAN input, gN/m2/s
+    real(r8), intent(inout) :: mtan(:) ! TAN pools for each age range. gN/m2
+    real(r8), intent(in) :: dt ! timestep, s
+    real(r8), intent(in) :: pools(:) ! age spans covered by each bin, seconds. size(mtan) = size(pools)
+    real(r8), intent(out) :: garbage ! TAN removed from the oldest pool. gN/m2
+
+    real(r8) :: flux_out(size(mtan))
+
+    flux_out = mtan / pools
+
+    ! new nitrogen
+    mtan(1) = mtan(1) + ndep * dt
+    ! transfer nitrogen from fresh to old pools
+    mtan = mtan - flux_out * dt
+    if (size(mtan) > 1) mtan(2:) = mtan(2:) + flux_out(:size(mtan)-1) * dt
+    ! provided that the oldest pool has wide enough age range, the amount transferred out
+    ! should be small.
+    garbage = flux_out(size(mtan)) * dt
+    
+  end subroutine age_pools_soil
+
+  subroutine age_pools_slurry(ndep, dt, water_slurry, tan_slurry, tan_soil, pools, garbage)
+    !
+    ! Age pools, modified for the 4-stage slurry model (only below-surface slurry ages).
+    !
+    implicit none
+    real(r8), intent(in) :: ndep ! flux of TAN input, gN/m2/s
+    real(r8), intent(in) :: dt ! timestep, s
+    ! water in slurry pool, on surface (1) and below surface (2) not including the background water content (theta)
+    real(r8), intent(in) :: water_slurry(:)
+    real(r8), intent(inout) :: tan_slurry  ! TAN in slurry pool, gN/m2
+    real(r8), intent(inout) :: tan_soil(:) ! TAN in soil pools, gN/m2
+    real(r8), intent(in) :: pools(:) ! age spans covered by each pool, including the S0 surface slurry. seconds.
+    real(r8), intent(out) :: garbage       ! garbage TAN, see above
+
+    real(r8) :: fract_slurry_soil, flux_to_soilpools
+    
+    fract_slurry_soil = water_slurry(2) / sum(water_slurry)
+    flux_to_soilpools = tan_slurry * fract_slurry_soil / pools(1)
+    tan_slurry = tan_slurry + (ndep - flux_to_soilpools) * dt
+
+    call age_pools_soil(flux_to_soilpools, dt, pools(2:), tan_soil, garbage)
+    
+  end subroutine age_pools_slurry
+
+  !************************************************************************************
+  ! Public functions for integrating the FAN model for one timestep.
+  
+  subroutine update_4pool(tg, ratm, theta, thetasat, precip, evap, qbot, watertend, runoff, tandep, tanprod, bsw, &
+       depth_slurry, poolranges, tanpools, Hconc, fluxes, residual, dt, dz_layer, pools_size, fluxes_size, status)
+    !
+    ! Evaluate fluxes and integrate states for a 4-stage slurry model with first pool
+    ! representing uninfiltrated slurry.
+    !
+    implicit none
```

Comment:
Here and in many other places: an `implicit none` statement at the top of a module means that you don't need one in every subroutine / function.